yaml.go

```diff
@@ -307,24 +308,69 @@ type yamlError struct {
 	err error
 }
 
+func (e *yamlError) Unwrap() error {
+	return e.err
+}
+
 func fail(err error) {
-	panic(yamlError{err})
+	panic(&yamlError{err})
 }
 
 func failf(format string, args ...interface{}) {
-	panic(yamlError{fmt.Errorf("yaml: "+format, args...)})
+	panic(&yamlError{fmt.Errorf("yaml: "+format, args...)})
+}
+
+// ParserError is each error with a source line position found by the parser.
+// Unlike UnmarshalErrors, it is only used for terminal failures.
```

Comment:
Slightly different wording:

```suggestion
// ParserError represents a fatal error encountered during the parsing phase.
// These errors typically indicate a syntax issue in the YAML document that
// prevents further processing.
```